packages/core/src/sync/index.ts

```diff
@@ -723,57 +711,67 @@ export const createSync = async (args: CreateSyncParameters): Promise<Sync> => {
             service: "sync",
             msg: `Finalized block for '${network.name}' has surpassed overall indexing checkpoint`,
           });
+          return;
```

Comment:
@typedarray Let me know what you think about this. Because we coupled events with raw block data, we must not flush either until it is known that the events have been processed. Another solution is to decouple these concepts, but I think the implementation of this will be harder to reason about and less maintainable.